keras_nlp/samplers/sampler.py

```diff
@@ -312,8 +317,9 @@ def sample(
         # The index of the last non-padding token in prompt. Since all sequences
         # are aligned to the right side, the index is the same for all.
         current_index = max_length - num_steps
+        original_padding_mask = tf.cast(tf.identity(mask), dtype=tf.int32)
```

Comment:
we compute this twice (above as well), should we just pass it through?

Reply:
It's fairly cheap to compute. I was doing this way because `sample` already has confusing arg list, would like to keep it shorter (though not much...)